vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java

```diff
@@ -518,6 +537,7 @@ private void ensureAttached() {
         ui.beforeClientResponse(ui, context -> {
             if (getElement().getNode().getParent() == null) {
                 ui.add(this);
+                ui.setChildComponentModal(this, isModal());
```

Comment:
This is not a problem if the opened dialog is a child of the modal dialog for instance if it is opened as:
```
        modalDialog.add(new NativeButton("open dialog", e -> {
            modalDialog.add(nonModalDialog);
            nonModalDialog.open();
        }));
```
But can this linking be made automatically somehow?

Reply:
Looks like a Dialog [gets automatically added](https://github.com/vaadin/flow-components/blob/master/vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java#L516-L524) to the UI on open in case it has no parent. Also ConfirmDialog, Notification and ContextMenu have similar logic.

Note that ConfirmDialog is also modal by default and should probably get the same fix.

Reply:
Added modality to ConfirmDialog and added test that having a non modal on a modal works if it is connected to the modal dialog. Created the ticket #2637 about automating the add to modal cases.

Reply:
If we merge this PR without the changes described in #2637 existing apps that rely on the current auto-add functionality will break.

Reply:
I'll implement vaadin/flow#12992 
When it is available should we add 2637 to this PR or make it on its own?

Reply:
Sounds good. Yeah, maybe it's better to include in this PR so `master` stays in "releasable" state.

Reply:
Now the overlay components with auto add to ui will add to the possible modal component so that they stay responsive.

Reply:
Tested the new revision and the issue above is now resolved.

There is, however, a behavioural change that may or may not be breaking. Since the sub dialog gets automatically added as a sub component of a modal parent dialog, it will automatically close when the parent dialog closes and automatically open when the parent dialog is re-opened.

https://user-images.githubusercontent.com/1222264/154486163-6f1c6163-bff4-491c-b9dc-6be9323c613c.mp4

Reply:
We discussed this internally and decided it's not a blocker for the PR.